### PR TITLE
Add support for STRING_MAP attributes.

### PR DESCRIPTION
**Pull request by geeknoid**
This change is [Reviewable](https://reviewable.io/reviews/istio/mixer/271)

### CODE REVIEW

**Comment by mandarjog** [comment]
@qiwzhang This PR adds support to send attribute of type `StringMap` `map<string,string>` which we expect request.headers / response.headers to be.
The mixer client code will need to support this data type.

**Comment by qiwzhang** [comment]
@mandarjog I will change the client to use it.

File: cmd/client/main.go [hunk 1]
```diff
@@ -47,6 +47,9 @@ type rootArgs struct {
 	// bytesAttributes is the list of name/value pairs of bytes attributes that will be sent with requests.
 	bytesAttributes string
 
+	// stringMapAttributes is the list of string maps that will be sent with requests
+	stringMapAttributes string ``
```
Comment (geeknoid) [inline cmd/client/main.go]: Not sure where that came from... Removed.

File: cmd/client/util.go [hunk 2]
```diff
@@ -99,6 +99,38 @@ func parseBytes(s string) (interface{}, error) {
...
+
+		k := pair[0:colon]
+		v := pair[colon+1:]
+		m[k] = v
+	}
+	return m, nil
+}
+
+// add to dictionary
+func add2Dict(dictionary map[int32]string, s string) int32 {
```
Comment (geeknoid) [inline cmd/client/util.go]: Given the context in which this is used, I didn't care about the redundancy in that case. But just to be anal, I added a little loop to discover and reuse existing dictionary entries.

File: cmd/client/util_test.go [hunk 3]
```diff
@@ -19,110 +19,137 @@ import (
...
+		{"v", 3.0},
+		{"w", true},
+		{"x", time.Date(2006, 1, 2, 15, 4, 5, 0, time.UTC)},
+		{"y", time.Duration(42) * time.Second},
+		{"z", []byte{0x98, 0x76}},
+		{"zz", map[string]string{"k3": "v3"}},
+	}
+
+	for _, r := range results {
+		v, found := attribute.Value(b, r.name)
```
Comment (geeknoid) [inline cmd/client/util_test.go]: I switched to t.Run in a few tests. Thanks.

**Comment by geeknoid** [comment]
PTAL.

**Pull request merged**